src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PropertyGridViewAccessibleObjectTests.cs

```diff
@@ -263,5 +263,25 @@ public void PropertyGridViewAccessibleObject_Parent_IsNotNull()
             ControlAccessibleObject accessibleObject = (ControlAccessibleObject)propertyGrid.GridViewAccessibleObject;
             Assert.NotNull(accessibleObject.Parent);
         }
+
+        [WinFormsFact]
+        public void PropertyGridViewAccessibleObject_EntryValuesListName_IsNotEmpty()
```

Comment:
Which codepath is this test covering?

Reply:
![image](https://user-images.githubusercontent.com/49272759/75333629-843b3480-5897-11ea-8e64-a839429a8212.png)
We have to ensure compliance with MAS responsibilities because an app developer doesn't have access to the Entry Values List AccessibleObject.
This test checks that this is done.

Reply:
We have added a new property `Name`, but the test (according to the name) exercises `EntryValuesListName`.
So my question stems from this - are we testing the new property? 

Also the property has several codepaths, e.g. 
* 1 here: `if (base.Name != null)`
* 3 here: `_owningGridViewListBox?.OwningPropertyGridView?.SelectedGridEntry?.PropertyName`
* 1 here: `propertyName ??= string.Empty`

Which one are we testing here? And do we have tests for the other paths?

Reply:
@RussKie 

> are we testing the new property?

Yes, we are testing propertyGridView.DropDownListBoxAccessibleObject.Name, which I implemented in this PR.

> 1 here: if (base.Name != null)

 `(base.Name != null)` means a custom accessible name was set.
It is checked here:
```cs
            string newValue = "Some text";
            propertyGridView.DropDownListBoxAccessibleObject.Name = newValue;
            listAccessibleName = propertyGridView.DropDownListBoxAccessibleObject.Name;
            Assert.Equal(newValue, listAccessibleName);
```

> - 4 here: _owningGridViewListBox?.OwningPropertyGridView?.SelectedGridEntry?.PropertyName
> - 1 here: propertyName ??= string.Empty

In any negative case, the string will be empty or null.
We could use `Assert.False(string.IsNullOrEmpty(listAccessibleName));` but I removed it as we decided in https://github.com/dotnet/winforms/pull/2843#discussion_r378684827.

> Which one are we testing here? And do we have tests for the other paths?

This test tests all code paths at the same time.

Reply:
> Yes, we are testing propertyGridView.DropDownListBoxAccessibleObject.Name, which I implemented in this PR.

Please rename the test to reflect what's being tested in the name, something like:
```suggestion
        public void PropertyGridView_DropDownListBoxAccessibleObject_Name_EntryValuesListName_IsExpected()
```

> This test tests all code paths at the same time.

Each unit test must test only a single code path.

> In any negative case, the string will be empty or null.

Correct, but each `?.` is a shortcut for `!= null`. If tomorrow one of these `?` gets accidentally deleted how do catch it?

Reply:
@RussKie, due the implementation changing, I haven't to make a lot of tests to check it. It needs 2 tests only now.
Thanks for your recommendations!

Reply:
@RussKie, due to the implementation changing, I haven't to make a lot of tests to check it. It needs 3 tests only now.
Thanks for your recommendations!